4C Integration.md

```diff
@@ -0,0 +1,262 @@
+* `Sets` and `Lists` are both sent as arrays, distinguished only by semantics
+* `Pages` are a wrapper object for paged endpoints, with these fields:
+    * **content**
+        * the current page of results
+    * **number**
+        * the current page number
+    * **totalPages**
+        * the total number of pages
+    * **first/last**
+        * `Booleans` indicating whether the current page is a terminal
+
+#### Entities
+
+All entities have `Ids`; they are unique, but not globally. There will never be duplicate ids within a single entity, but there may be across entities.
+
+Not all fields are documented here; the ones left off are safe to ignore.
+
+##### `AvailablilityGroup`
+
+Represents a group of avails with the same asset
+
+* asset: `Set<Attribute>`
+* avails: `List<Availablility>`
+* marketId: `Id`
+
+##### `Availablility`
+
+* start: `Date`
+* end: `Date`
+* qty: `Number`
+* price: `Currency`
+
+##### `Attribute`
+
+* type: `AttributeType`
+* value: `String`
+
+##### `AttributeType`
+
+* name: `String`
+
+##### `Campaign`
+
+* name: `String`
+* advertiser: `String`
+* brand: `String`
+* flightStartDate: `Date`
+* flightEndDate: `Date`
+
+##### `OrderGroup`
+
+* qty: `Number`
+    * Quantity of units desired
+* flightStartDate: `Date`
+* flightEndDate: `Date`
+* marketId: `Id`
+    * Market the `OrderGroup` will eventually be submitted into
+* price: `Currency`
+* asset: `Set<Attribute>`
+
+##### `Deal`
+
+* network: `String`
+* unitId: `String`
+* broadcastDate: `Date`
+* airTime: `Time`
+* status: `DealStatus`
+* revenueStream: `RevenueStream`
+* marketplace: `Marketplace`
+* dealSalesType: `DealSalesType`
+* unitStatus: `UnitStatus`
+* logStatus: `LogStatus`
+* spotType: `SpotType`
+* guaranteed: `Boolean`
+* onLog: `Boolean`
+* agency: `String`
+* advertiser: `String`
+* brand: `String`
+* category: `String`
+* grossDollars: `Currency`
+* rateCardDollars: `Currency`
+* asset: `Set<Pair<String, String>>`
+
+##### `OrderQuery`
+
+* start: `Date`
+* end: `Date`
+* attributes: `Set<Id>`
+* role: `Enum<MarketParticipantType>`
+    * set this to `ADVERTISER`
+* pageIndex: `Number`
+
+##### `MatchQuery`
+
+* start: `Date`
+* end: `Date`
+* byMatchDate: `Boolean`
+    Indicates the query mode, ie if the dates refer to
+    * match date
+    * asset date
+* pageIndex: `Number`
+
+
+#### Endpoints
+
+##### Market Query
+
+Query avails for a given `Asset` in a `Market`
+
+`GET /market/sells`
+Query params: `OrderQuery`
+
+Returns: `Page<AvailablilityGroup>`
+
+##### Attribute Query
+Query all known attributes
+
+`GET /attr`
+Returns: `Attribute`
+
+##### Campaign Creation
+Create a new `Campaign`
+
+`POST /campaign`
+Request body: `Campaign`
+
+##### Buy Order Submission
+
+Add a new `OrderGroup` to a `Campaign`
+
+`POST /campaign/{campId}/orders`
+Path variables:
+* campId: `Id`
+`Campaign` being added to
+
+Request body: `OrderGroup`
+
+##### Buy Order Status Query
```

Comment:
status is a little misleading here? maybe 'Buy Order Query' and the next line is 'Query for the current version of your OrderGroup'?